tests/plugins/link/anchor.js

```diff
@@ -79,6 +79,46 @@
 					assert.areSame( anchor, dialog.getModel( editor ) );
 				} );
 			} );
+		},
+
+		// (#3863)
+		'test preserve duplicate anchors after editing word with custom style': function() {
+			var editor = this.editor,
+				bot = this.editorBot,
+				html = '<p><a id="test" name="test"><strong>text</strong></a></p>';
+
+			bot.setData( html, function() {
+				var range = editor.createRange(),
+					anchor = editor.editable().findOne( '#test' ),
+					textNode = anchor.getChild( 0 );
+
+				range.selectNodeContents( textNode );
+				range.select();
+
+				bot.dialog( 'anchor', function( dialog ) {
+					dialog.setValueOf( 'info', 'txtName', 'duplicate-test' );
+					dialog.getButton( 'ok' ).click();
+
+					assert.areSame( anchor, dialog.getModel( editor ) );
+				} );
+
+			} );
```

Comment:
This test could be written the same way as [the 2nd one you have added](https://github.com/ckeditor/ckeditor4/pull/4733/files#diff-79b36452737f4e4748cde2f72e7ab1eb42745722781f4214899d2fa3ced4584fR109) (which is simpler I guess). Was there any particular reason for doing it like this?